config/main.py

```diff
@@ -527,6 +527,16 @@ def load_minigraph():
     """Reconfigure based on minigraph."""
     log_info("'load_minigraph' executing...")
 
+    # get the device type
+    command = "{} -m -v DEVICE_METADATA.localhost.type".format(SONIC_CFGGEN_PATH)
+    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    device_type, err = proc.communicate()
+    if err:
+        click.echo("Could not get the device type from minigraph, exiting")
+        sys.exit(1)
```

Comment:
Do we have legitimate case where device type is not available in minigraph?

Reply:
From a cautious perspective. I am leaning towards letting this error go and enable pfcwd. There might be a default or test mingiraph missing device type?

Reply:
will make that change